### ISSUE TITLE
Automate DNS on approved PRs

### ISSUE THREAD

**Issue by thockin** [kubernetes/k8s.io#507]
When a DNS PR is approved, a human has to run the DNS syncer.  We should automate that.  It can run in aaa, watch the git repo, (use workload identity to access DNS?) and needs to send results somewhere.

**Comment by thockin** [comment]
Dup of #239 

Stealing the context: 

Followup from #151

At the moment, updating DNS still requires humans to run commands. The ideal state is PR-driven:

DNS requests are made via PR
OWNERS files dictate who has authority to approve a given DNS request
When then PR merges, something deploys the change
https://github.com/kubernetes/k8s.io/tree/master/dns#todo outlines some potential tasks

The model we've used elsewhere is:

Build an image and store it somewhere in gcr.io
Use prow jobs that run in a "trusted" cluster (has credentials to do things, so don't let untrusted code run in it)
Use postsubmits to deploy immediately
Use periodics to reconcile state
Which would mean this may be blocked on:

community owned gcr.io repo for k8s-infra
community owned prow
community owned trusted prow cluster

**Comment by palnabarun** [comment]
Hi @thockin, I would like to help out with this.

**Comment by palnabarun** [comment]
What I have understood is currently, after any changes are made to the DNS configuration, someone who has the required credentials runs `make check` and `make push`.

To automate this, we can run `make check` whenever any PR is submitted to change the DNS configuration. And when the PR is approved and merged, we can run `make push` to push the final DNS changes. PR approval as said above is restricted to OWNERS inside `k8s.io/dns`.

**Comment by spiffxp** [comment]
/unassign @palnabarun 
if you're still working on this, please /assign yourself and let us know what's new
/assign @bartsmykla 
Trying to implement as a prowjob using the existing trusted-cluster approach used by container image promoter.  Currently stuck on how to run octodns without a service account key file so that it can automatically pickup what service account is provided via workload identity

WIP: https://github.com/kubernetes/k8s.io/tree/master/dns#running-automated

**Comment by spiffxp** [comment]
/area dns
/area cluster-infra
/remove-help
since @bartsmykla is working on this

**Comment by spiffxp** [comment]
/close
I'm closing this in favor of https://github.com/kubernetes/k8s.io/issues/783 where @bartsmykla has been describing the plan and work

**Comment by bartsmykla** [comment]
Ah, sorry, I forgot about this issue and created the new one
